pkg/nsx/services/realizestate/realize_state.go

```diff
@@ -68,9 +90,26 @@ func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, inte
 						errMsg = append(errMsg, *alarm.Message)
 					}
 				}
-				return NewRealizeStateError(fmt.Sprintf("%s realized with errors: %s", entityType, errMsg))
+				if !(exists && entity) {
+					if entityType != "" && entityType == *result.EntityType || entityType == "" {
+						log.Error(errors.New("realize state error"), "entityType", *result.EntityType, "path", intentPath)
+					}
+					entitiesMsg[*result.EntityType] = fmt.Sprintf("%s realized with errors: %s", *result.EntityType, errMsg)
+				}
```

Comment:
I'm a little bit confused here, why we need to introduce entitiesMsg/entitiesStatus here instead of just returning the error when we found any one of the entity is not realized?

Reply:
there may be multiple items for one resource, that one of them failed, but the other of them realized. If one of them realized, we thought it's realized no matter the failed one